src/main/java/homework08/BookCrossingLibrary.java

```diff
@@ -0,0 +1,52 @@
+package homework08;
+
+import java.util.concurrent.BlockingQueue;
+
+public class BookCrossingLibrary implements Runnable{
+	private BlockingQueue<Book> books;
+	private int bookLimit = 10;
+
+	public BookCrossingLibrary(int bookLimit) {
+		this.bookLimit = bookLimit;
+	}
+
+	public synchronized void bringBook(Book item) throws InterruptedException {
+		while (this.books.size() == this.bookLimit) {
+			wait();
```

Comment:
I dont understand why you need wait and notify

Reply:
You have now two synhronize methods. On level BookCrossingLibrary  you can use it same time only one